### ISSUE TITLE
Support package dependencies

### ISSUE THREAD

**Issue by lachmanfrantisek** [packit/packit-service#1993]
A common use case across the whole Packit service platform is to support dependencies between packages. We need to find a good way how to define relations between packages and how to work with them.

Notes:
* In upstream, this can be either a mono-repo (more packages per one git repo) or a relation that can spread multiple git repositories.
    *  Work on Monorepo is being done here: #1897 (the part of that is introducing a "package" concept into the configuration => we can use it.
* For downstream, the package is represented by a single git repository.
    * But the realisations are still important.

Here are the action items to make it work:
* [ ] #1994
* [ ] #1995
* [ ] Implementation steps will be added once the research and plan is done

See also https://github.com/packit/packit/issues/1903

**Comment by lachmanfrantisek** [comment]
Closing in favour of #2105
